polkadot/runtime/parachains/src/paras_inherent/mod.rs

```diff
@@ -430,6 +434,9 @@ impl<T: Config> Pallet<T> {
 			T::DisputesHandler::filter_dispute_data(set, post_conclusion_acceptance_period)
 		};
 
+		// Filter out backing statements from disabled validators
+		let statements_were_dropped = filter_backed_statements::<T>(&mut backed_candidates)?;
```

Comment:
this should probably go into `sanitize_backed_candidates`

Reply:
If we want to keep [this check](https://github.com/paritytech/polkadot-sdk/pull/1863#discussion_r1356858614) we'll need yet another function filtering backing votes.

Reply:
Considering the rest of your comments - I agree. Filtering on too many places is a bad idea. I'll move it to `sanitize_backed_candidates`